File: Xenon_VSCode/Xenon/RetroEngine.ts
import gsCVector = require("Vector");
import CActor = require("Actor");
import CEngine = require("Engine");
import CActorInfoList = require("ActorInfoList");
import enums = require("Enums");

class CRetroEngine extends CEngine {
    private m_direction: enums.RetroDirection;
    //private m_actorInfo: CActorInfoList;

    act: CActor;
    m_name = "RetroEngine";

    public CRetroEngine(listOfActors: CActorInfoList) {
        this.m_direction = enums.RetroDirection.RETRO_NW;
        this.m_actorInfo = listOfActors;
    }

    //-------------------------------------------------------------

    public Draw(ctx: CanvasRenderingContext2D) {
        //if (getOwner() &&  getOwner().getActorInfo().m_type == ACTOR_TYPE_SHIP &&
        //    ((CShip *) getOwner())->isCloaked())
        //    return true;

        if (this.m_thrust > 0) {
            switch (this.m_direction) {
                case enums.RetroDirection.RETRO_NW:
                    this.animations(enums.AnimationMode.ANIMATE_LOOP, 0, 2);
                    break;
                case enums.RetroDirection.RETRO_NE:
                    this.animations(enums.AnimationMode.ANIMATE_LOOP, 2, 2);
                    break;
                case enums.RetroDirection.RETRO_SW:
                    this.animations(enums.AnimationMode.ANIMATE_LOOP, 4, 2);
                    break;
                case enums.RetroDirection.RETRO_SE:
                    this.animations(enums.AnimationMode.ANIMATE_LOOP, 6, 2);
                    break;
            }

            super.Draw(ctx);
        }

        return true;
    }

    //-------------------------------------------------------------

    public setDirection(direction: enums.RetroDirection) {
        this.m_direction = direction;
    }

    //-------------------------------------------------------------

    public getActorInfo() {
        this.m_actorInfo = this.m_scene.GetlistOfActors();
        return this.m_actorInfo.GetActorInfoListItem(enums.ActorInfoType.INFO_RETRO_ENGINE);

    }
}
export = CRetroEngine;

File: Xenon-Original_C++_Code/xenon/source/retroengine.cpp
//-------------------------------------------------------------
//
// Class:	CRetroEngine
//
// Author:	John M Phillips
//
// Started:	06/05/00
//
// Base:	CEngine
//
// Derived:	None
//
//-------------------------------------------------------------

#include "game.h"

//-------------------------------------------------------------

CRetroEngine::CRetroEngine()
{
	m_direction = RETRO_NW;
}

//-------------------------------------------------------------

CRetroEngine::~CRetroEngine()
{
}

//-------------------------------------------------------------

bool CRetroEngine::draw()
{
	if (getOwner() &&
		getOwner()->getActorInfo().m_type == ACTOR_TYPE_SHIP &&
		((CShip *) getOwner())->isCloaked())
		return true;

	if (m_thrust > 0) {
		switch (m_direction) {
			case RETRO_NW:	animate(ANIMATE_LOOP,0,2);	break;
			case RETRO_NE:	animate(ANIMATE_LOOP,2,2);	break;
			case RETRO_SW:	animate(ANIMATE_LOOP,4,2);	break;
			case RETRO_SE:	animate(ANIMATE_LOOP,6,2);	break;
			}
		CActor::draw();
		}

	return true;
}

//-------------------------------------------------------------

void CRetroEngine::setDirection(RetroDirection direction)
{
	m_direction = direction;
}

//-------------------------------------------------------------
